pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -108,12 +119,59 @@ func (c *LatencyProfileController) sync(ctx context.Context, syncCtx factory.Syn
 	uniqueRevisions := []int32{}
 	for _, nodeStatus := range operatorStatus.NodeStatuses {
 		revision := nodeStatus.CurrentRevision
+
+		// during day-0: pre-empt
+		if revision == 0 {
+			_, err = c.updateStatus(
+				ctx,
+				false, false,
+				reasonClusterInit,
+				"one or more static pod revision(s) are unknown",
+			)
+			return err
+		}
+
 		if !revisionMap[revision] {
 			revisionMap[revision] = true
 			uniqueRevisions = append(uniqueRevisions, revision)
 		}
 	}
 
+	// In case a checkProfileRejection func was used, status will be updated accordingly
+	// ("" profile is also default profile, so "" -> anotherProfile: rejection can be handled)
+	if c.checkProfileRejectionFn != nil {
+		isRejected, rejectMsg, err := c.checkProfileRejectionFn(configNodeObj.Spec.WorkerLatencyProfile, uniqueRevisions)
+		if err != nil {
+			return err
+		}
+		// if profile transition is rejected, set status to progress=Unknown, complete=Unknown
+		// and degrade operator with suitable error message
+		if isRejected {
+			_, err = c.updateStatus(
+				ctx,
+				false, false, // not progressing, not complete
```

Comment:
Updated!

Reply:
ok
- rejectMsg is not propagated to the status
- do you see any circumstance where we would need `ConditionUnknown` ?

Reply:
> - rejectMsg is not propagated to the status

It is indirectly being propagated via the error message, so when the error is raised, it degrades the controller and the `WorkerLatencyProfileDegraded` condition in the status has the `rejectMsg` in it's message.

```yaml
status:
    conditions:
    - lastTransitionTime: "2022-06-14T08:20:44Z"
      reason: ProfileUpdateProhibited
      status: "False"
      type: WorkerLatencyProfileProgressing
    - lastTransitionTime: "2022-06-14T08:18:08Z"
      reason: ProfileUpdateProhibited
      status: "False"
      type: WorkerLatencyProfileComplete
--
    - lastTransitionTime: "2022-06-14T08:20:44Z"
      message: rejected update from "" to "LowUpdateSlowReaction" latency profile
        as extreme profile transition is unsupported, please set nodes.config.openshift.io/cluster
        object to workerLatencyProfile "" (or any other profile apart from rejected
        "LowUpdateSlowReaction") in order to recover from this error
      reason: SyncError
      status: "True"
      type: WorkerLatencyProfileDegraded
```
---
> - do you see any circumstance where we would need ConditionUnknown ?

Din't come across any situation where I feel it should be unknown, the reasoning behind it would be we actually manage the state of the operands (updating latency profile arg value pairs) through config observer(s) which is already managed and not manual which helps us always have some information about the current state of the latency profile through the mix of status(es) we crafted.